### PR TITLE
Helm chart refactor

### PR DESCRIPTION
**Pull request by Dean-Coakley**
This PR aims to realign the helm chart with current best practices.

### Proposed changes
* Add documentation to values.yaml
* Add helper template to reduce template repetition
* Update to latest kubernetes label standards
* Add icon to Chart.yaml
* Fix README.md to have consistant format

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [x] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/master/CONTRIBUTING.md) doc
- [x] I have checked that all unit tests pass after adding my changes
- [x] I have updated necessary documentation
- [x] I have rebased my branch onto master
- [x] I will ensure my PR is targeting the master branch and pulling from my branch from my own fork

### CODE REVIEW

File: deployments/helm-chart/values.yaml [hunk 1]
```diff
@@ -1,63 +1,176 @@
...
+
   image:
+    ## The image repository of the Ingress controller.
     repository: nginx/nginx-ingress
+
+    ## The tag of the Ingress controller image.
     tag: "edge"
+
+    ## The pull policy for the Ingress controller image.
+    ## Ref: http://kubernetes.io/docs/user-guide/images/#updating-images
```
Comment (pleshakov) [inline deployments/helm-chart/values.yaml]: this is not present in the doc. please remove

File: deployments/helm-chart/values.yaml [hunk 2]
```diff
@@ -1,63 +1,176 @@
...
-    # secret: <namespace>/<secret_name>
+
+    ## The secret with a TLS certificate and key for the default HTTPS server. The value must follow the following format: <namespace>/<name>.
+    ## Used as an alternative to specifiying a certifcate and key using controller.defaultTLS.cert and controller.defaultTLS.key parameters.
+    secret: #<namespace>/<secret_name>
+
+  ## The node selector for pod assignment for the Ingress controller pods.
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
```
Comment (pleshakov) [inline deployments/helm-chart/values.yaml]: not present in the doc. please remove

File: deployments/helm-chart/values.yaml [hunk 3]
```diff
@@ -1,63 +1,176 @@
...
+
+  ## The termination grace period of the Ingress controller pod.
   terminationGracePeriodSeconds: 30
+
+  ## The tolerations of the Ingress controller pods.
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
   tolerations: []
+
+  ## The affinity of the Ingress controller pods.
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity
```
Comment (pleshakov) [inline deployments/helm-chart/values.yaml]: not present in the doc. please remove

File: deployments/helm-chart/values.yaml [hunk 4]
```diff
@@ -1,63 +1,176 @@
...
+
+      ## The HTTPS port of the Ingress controller service.
       port: 443
+
+      ## The custom NodePort for the HTTPS port. Requires controller.service.type set to NodePort.
       nodePort: ""
+
   serviceAccount:
+    ## Creates a service account. Required for RBAC.
+    create: true
```
Comment (pleshakov) [inline deployments/helm-chart/values.yaml]: this is not used in the manifests. consider removing it

File: deployments/helm-chart/values.yaml [hunk 5]
```diff
@@ -1,63 +1,176 @@
...
+
+    ## The name of the service account of the Ingress controller pods. Used for RBAC.
     name: nginx-ingress
+
+    ## The names of the secrets containing docker registry credentials.
+    ## Ref: https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/
```
Comment (pleshakov) [inline deployments/helm-chart/values.yaml]: not present in the doc. please remove.

File: deployments/helm-chart/values.yaml [hunk 6]
```diff
@@ -1,63 +1,176 @@
...
+
+  ## The node selector for pod assignment for the Ingress controller pods.
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#nodeselector
   nodeSelector: {}
+
+  ## The termination grace period of the Ingress controller pod.
   terminationGracePeriodSeconds: 30
+
+  ## The tolerations of the Ingress controller pods.
+  ## Ref: https://kubernetes.io/docs/concepts/configuration/taint-and-toleration/
```
Comment (pleshakov) [inline deployments/helm-chart/values.yaml]: not present in the doc. please remove

File: deployments/helm-chart/templates/_helpers.tpl [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+{{- define "nginx-ingress.name" -}}
+{{- .Chart.Name | trunc 63 | trimSuffix "-" -}}
+{{- end -}}
+
+{{/*
+Create labels
+*/}}
+{{- define "nginx-ingress.labels" -}}
+app.kubernetes.io/name: {{ include "nginx-ingress.name" . }}
+helm.sh/chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
```
Comment (pleshakov) [inline deployments/helm-chart/templates/_helpers.tpl]: hm. why `replace "+" "_"`?

File: deployments/helm-chart/templates/_helpers.tpl [hunk 8]
```diff
@@ -0,0 +1,18 @@
+{{/* vim: set filetype=mustache: */}}
```
[the remark this answers is not in the document]
Comment (Dean-Coakley) [reply deployments/helm-chart/templates/_helpers.tpl]: I do

File: deployments/helm-chart/values.yaml (hunk 4 above)
Comment (Dean-Coakley) [reply deployments/helm-chart/values.yaml]: Nice catch, will do.

**Pull request merged**